src/main/java/org/mockito/internal/creation/bytebuddy/InlineByteBuddyMockMaker.java

```diff
@@ -300,19 +314,41 @@ public MockHandler getHandler(Object mock) {
     public void resetMock(Object mock, MockHandler newHandler, MockCreationSettings settings) {
         MockMethodInterceptor mockMethodInterceptor =
                 new MockMethodInterceptor(newHandler, settings);
-        mocks.put(mock, mockMethodInterceptor);
-        if (mock instanceof MockAccess) {
-            ((MockAccess) mock).setMockitoInterceptor(mockMethodInterceptor);
+        if (mock instanceof Class<?>) {
+            Map<Class<?>, MockMethodInterceptor> interceptors = mockedStatics.get();
```

Comment:
I am not really following this refactoring. Could you elaborate on why this is necessary? Would it maybe be possible to separate these changes into its own PR to ease reviewing?

Reply:
It's easier when looking at the raw file, the changefile cuts parts of the interesting segment. Basically, to avoid breaking the existing implementation, static mocks are represented by the `Class` value for the static mock. This is also done in the listerners etc. what is possible since it is currently impossible to mock a `Class` instance, even with the inline-mock maker.

We cannot store static mocks in the same concurrent map as the regular mocks since they are thread local. Therefore, we always need to make a distinction if a mock is an instance of `Class` or not to see if a static or a regular mock is requested. The advantage of this solution is that the current API can mainly remain untouched, keeping the changeset to a minimum.